distributions/otelcol-k8s/config.yaml

```diff
@@ -0,0 +1,46 @@
+# To limit exposure to denial of service attacks, change the host in endpoints below from 0.0.0.0 to a specific network interface.
+# See https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md#safeguards-against-denial-of-service-attacks
+
+extensions:
+  health_check:
+
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: 0.0.0.0:4317
+      http:
+        endpoint: 0.0.0.0:4318
+
+  # Collect own metrics
+  prometheus:
+    config:
+      scrape_configs:
+        - job_name: 'otel-collector'
+          scrape_interval: 10s
+          static_configs:
+            - targets: ['0.0.0.0:8888']
```

Comment:
We kind of touched on this topic while having a different discussion at KubeCon and I think we shouldn't encourage people to have collectors scraping its own metrics.

Reply:
I can understand why, but often if you want to use the collector to collect Prometheus metrics to begin with, it's simpler to just add a ServiceMonitor for the collector itself (the operator automates this too) instead of treating collector metrics as a special case.

I don't object to removing this from the default config, just a general comment.

Reply:
Updated